Cannot execute the test suite when Git Signing commits is enabled by default

Hi, I have configured my local Git to sign automatically all the commits for all my repositories, see https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits. When this facility is enabled, the Prospero tool cannot be built by running the test suite.

This is an example of the errors you get:

```
[ERROR] Tests run: 5, Failures: 0, Errors: 5, Skipped: 0, Time elapsed: 0.322 s <<< FAILURE! - in org.wildfly.prospero.installation.LocalInstallationHistoryTest
[ERROR] getHistoryOfInstallations(org.wildfly.prospero.installation.LocalInstallationHistoryTest)  Time elapsed: 0.245 s  <<< ERROR!
org.wildfly.prospero.api.exceptions.MetadataException: Unable to write history of installation
        at org.wildfly.prospero.installation.git.GitStorage.record(GitStorage.java:92)
        at org.wildfly.prospero.api.InstallationMetadata.recordProvision(InstallationMetadata.java:221)
        at org.wildfly.prospero.test.MetadataTestUtils.createInstallationMetadata(MetadataTestUtils.java:76)
        at org.wildfly.prospero.installation.LocalInstallationHistoryTest.mockInstallation(LocalInstallationHistoryTest.java:137)
        at org.wildfly.prospero.installation.LocalInstallationHistoryTest.getHistoryOfInstallations(LocalInstallationHistoryTest.java:65)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
        at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: org.eclipse.jgit.api.errors.ServiceUnavailableException: Signing service is not available
        at org.eclipse.jgit.api.CommitCommand.sign(CommitCommand.java:290)
        at org.eclipse.jgit.api.CommitCommand.call(CommitCommand.java:247)
        at org.wildfly.prospero.installation.git.GitStorage.record(GitStorage.java:86)
        ... 35 more
```

To workaround the issue, I have disabled the automatic commit signing by using:

`git config --global commit.gpgsign false`

And then, the test suite runs without any issues.

--- a/prospero-common/src/main/java/org/wildfly/prospero/installation/git/GitStorage.java
+++ b/prospero-common/src/main/java/org/wildfly/prospero/installation/git/GitStorage.java
@@ -17,6 +17,7 @@
 
 package org.wildfly.prospero.installation.git;
 
+import org.eclipse.jgit.lib.StoredConfig;
 import org.wildfly.prospero.api.InstallationMetadata;
 import org.wildfly.prospero.api.exceptions.MetadataException;
 import org.wildfly.prospero.api.SavedState;
@@ -186,6 +187,9 @@ public class GitStorage implements AutoCloseable {
         Git git;
         if (!base.resolve(".git").toFile().exists()) {
             git = Git.init().setDirectory(base.toFile()).call();
+            final StoredConfig config = git.getRepository().getConfig();
+            config.setBoolean("commit", null, "gpgsign", false);
+            config.save();
         } else {
             git = Git.open(base.toFile());
         }

